src/deploy/functions/runtimes/node/parseTriggers.ts

```diff
@@ -216,19 +221,27 @@ export function addResourcesToBackend(
       triggered = {
         eventTrigger: {
           eventType: annotation.eventTrigger!.eventType,
-          eventFilters: [
-            {
-              attribute: "resource",
-              value: annotation.eventTrigger!.resource,
-            },
-          ],
           retry: !!annotation.failurePolicy,
+          eventFilters: [],
         },
       };
+      if (annotation?.eventTrigger?.channel) {
+        triggered.eventTrigger.channel = resolveChannelName(
+          projectId,
+          annotation?.eventTrigger?.channel
+        );
+      }
+
+      if (annotation.eventTrigger!.resource) {
+        triggered.eventTrigger.eventFilters!.push({
+          attribute: "resource",
+          value: annotation.eventTrigger!.resource,
+        });
+      }
```

Comment:
IIUC, custom events are only supported in Cloud Functions Gen2.

We no longer use `parseTrigger` to read trigger definitions for CF3v2 deploys - we exclusively rely on container contract to do it. See https://github.com/firebase/firebase-tools/blob/master/src/deploy/functions/runtimes/node/index.ts#L127-L141.

This code should probably be written in `src/deploy/functions/runtimes/discovery/v1alpha1.ts` instead.

I suspect this change worked and you were able to deploy your custom event fns from your local dev machine... My guess is that either

1) your locally built tarball specifies an old version of the SDK
2) you haven't enabled cloudfunctionsv2 preview flag

Reply:
Done!